### ISSUE TITLE
DrawImageRect -> DrawRect

### ISSUE THREAD

**Issue by nsymtex** [Regal-Internet-Brothers/mojoinmojo2#1]
Possible to handle this? mojo2 wants DrawRect

DrawImageRect(coin, 200, 100, 0, 0, coin.Width, coin.Height, 0)

where coin:Image = just any image.

**Comment by ImmutableOctet** [comment]
Just fixed this, it was an oversight. Thanks for the help. I'll look into turning reflection on, so I can see if there's any other issues. Just pull the latest version, and it *should* compile. (Ideally)

**Comment by ImmutableOctet** [comment]
I pushed another fix, everything compiles with reflection now. That doesn't mean everything functions accurately, but it'll at least build.

**Comment by nsymtex** [comment]
IOnLoadImageComplete is missing next

asyncloaders.monkey / async in general

**Comment by ImmutableOctet** [comment]
Just added support for 'IOnLoadImageComplete' / 'LoadImageAsync'. It's even threaded, so it should function the same. There's also 'MOJOINMOJO2_THREADED_ASYNC', which is a boolean configurable with the preprocessor. Enable it to use proper threads, disable it to make a direct call to the underlying implementation. **By default, this is enabled.**

**Comment by ImmutableOctet** [comment]
Just finished up a proxy to Mojo's 'IOnLoadSoundComplete' / 'LoadSoundAsync'. Let me know if there's anything else I'm missing.

**Comment by nsymtex** [comment]
Compiling seems to be all working now for what we were testing on

mojoinmojo2.LoadString seems to be non-functional

**Comment by ImmutableOctet** [comment]
@nsymtex: I should probably disclose that 'mojoinmojo2' is a Mojo ***graphics*** API layer. Everything related to normal Mojo excluding graphics is routed through 'mojo' normally, which is the same as what Mojo 2 does. So, since this doesn't wrap 'mojo.app' in any way, just provides an automatic/public import, the name is probably still 'mojo.LoadString', 'app.LoadString', or 'mojo.LoadString'. To my knowledge, the only way to fix this is to add aliases to every single function. Which is possible, and will probably be added to the bucket list, but isn't a priority. The only other way I can think of to get around this is to remove import-compatibility for 'app'.

So, basically, use something like 'mojo.LoadString', or just 'LoadString'.

**Comment by nsymtex** [comment]
Using  'mojo.LoadString' and 'LoadString' did not work sadly. Will try more things.

**Comment by nsymtex** [comment]
The LoadString problem - for some reason when working with mojo2 fnt ansi encoded files are not working... renaming to txt works. Still look at why this is happening. It may be a mistake we have done somewhere.

**Comment by ImmutableOctet** [comment]
Sounds like an unrelated problem. Have you tried this?
> #TEXT_FILES += "*.FORMATHERE"

Build settings get reset when you change Monkey versions, unless you configure your project with a separate file.

**Comment by nsymtex** [comment]
That fixed it, thank you!

Do you own Regal Internet Brothers? Are there game projects you are working on?

**Comment by ImmutableOctet** [comment]
I do. As far as games go, there's been a few things I've worked on, mainly prototypes. Nothing publicly distributed. Most of the work I've wanted to publicize has been uploaded on GitHub already.
